DscConfigurationData/Public/Get-DscConfigurationData.ps1

```diff
@@ -45,11 +45,11 @@ function Get-DscConfigurationData
         {
             'NameFilter' {
                 Write-Verbose "Filtering for nodes with the Name $Name"
-                $script:ConfigurationData.AllNodes = $script:ConfigurationData.AllNodes.Where({$_.Name -like $Name})
+                [hashtable[]]$script:ConfigurationData.AllNodes = $script:ConfigurationData.AllNodes.Where({$_.Name -like $Name})
```

Comment:
Cast to hashtable array as dsc was bombing out when compling mof if the filter only returned one node since AllNodes needs to be a collection.